### PR TITLE
Upgrade to Mongo 3.6

### PR DESCRIPTION
**Pull request by melvinsoft**

### CODE REVIEW

**Comment by thraxil** [comment]
Exciting. The commit notes say "This version of pymongo supports MongoDB 2.x all the way up to 4.2". We might even be able to move to 4.x in the foreseeable future.

**Comment by melvinsoft** [comment]
@OmarIthawi Thanks for the review, I added the test you requested but then I deleted one: `lms/djangoapps/verify_student/management/commands/tests/test_send_verification_expiry_email.py` since is testing new code we don't have yet, and more intended for python 3 rather than 2. 

But tests are still failing, new fails are related to Access Control Backend: 
```
py27-lms run-test: commands[3] | lms/lib/tests/test_access_control_backends.py openedx/core/djangoapps/appsembler openedx/core/djangoapps/site_configuration/tests/test_tahoe_changes.py openedx/core/djangoapps/user_api/accounts/tests/test_utils.py::CompletionUtilsTestCase
ERROR: InvocationError for command could not find executable lms/lib/tests/test_access_control_backends.py
```
I invested about three hours until I reached this point, can you please take a quick look? If not we should move forward without the new tests, and do it as a follow up work.

What do you think?

**Comment by OmarIthawi** [comment]
Thanks for trying that much @melvinsoft! I think we should skip the broken tests and keep the others.

### TL; DR;
In general I think in tests for the `edx-platform` we probably want to take things slowly and steer clear from rabbit holes.

Even in the best case scenario, we don't want all tests to run, maybe 99% at max, this is to make it possible for us to deviate from Open edX in a healthy way until we make plugins or APIs.

Thank you again!

File: tox.ini [hunk 1]
```diff
@@ -122,10 +122,13 @@ commands =
         lms/djangoapps/courseware/tests/test_access_control_backends_integration.py \
         lms/djangoapps/grades/tests/integration/test_events.py \
         lms/djangoapps/instructor/tests/test_certificates.py::CertificatesInstructorApiTest \
+        lms/djangoapps/verify_student/tests/test_services.py
```
Comment (OmarIthawi) [inline tox.ini]:
```suggestion
        lms/djangoapps/verify_student/tests/test_services.py  \
```

File: tox.ini [hunk 2]
```diff
@@ -122,10 +122,13 @@ commands =
...
         lms/lib/tests/test_access_control_backends.py \
         openedx/core/djangoapps/appsembler \
         openedx/core/djangoapps/site_configuration/tests/test_tahoe_changes.py \
         openedx/core/djangoapps/user_api/accounts/tests/test_utils.py::CompletionUtilsTestCase
+        common/lib/xmodule/xmodule/modulestore/tests/test_mongo_call_count.py
```
Comment (OmarIthawi) [inline tox.ini]:
```suggestion
        common/lib/xmodule/xmodule/modulestore/tests/test_mongo_call_count.py \
```

File: tox.ini [hunk 3]
```diff
@@ -122,10 +122,13 @@ commands =
         lms/djangoapps/courseware/tests/test_access_control_backends_integration.py \
         lms/djangoapps/grades/tests/integration/test_events.py \
         lms/djangoapps/instructor/tests/test_certificates.py::CertificatesInstructorApiTest \
+        lms/djangoapps/verify_student/tests/test_services.py
         lms/lib/tests/test_access_control_backends.py \
         openedx/core/djangoapps/appsembler \
         openedx/core/djangoapps/site_configuration/tests/test_tahoe_changes.py \
         openedx/core/djangoapps/user_api/accounts/tests/test_utils.py::CompletionUtilsTestCase
```
Comment (OmarIthawi) [inline tox.ini]:
```suggestion
        openedx/core/djangoapps/user_api/accounts/tests/test_utils.py::CompletionUtilsTestCase \
```

**Comment by melvinsoft** [comment]
@OmarIthawi Thanks for all the help here. I'm merging this now.

**Comment by melvinsoft** [comment]
@OmarIthawi Why do you think is gonna break devstack? because the `authSource`  and `replicaSet` parameters in the connection?
